client/src/pages/Balance/Checkout.jsx

```diff
@@ -62,7 +49,8 @@ const Checkout = ({
   const { user } = useContext(UserContext);
   const stripe = useStripe();
   const elements = useElements();
-  const [error, setError] = useState(null);
+  const [errorSnackbar, setErrorSnackbar] = useState({ open: false, message: '' });
+  const [successSnackbar, setSuccessSnackbar] = useState({ open: false, message: '' });
```

Comment:
I mentioned this in Frank's PR too, since you are using similar logic here, but:

You could probably consolidate both of these snackbars into one snackbar by adding a field called severity (and that could be either success or error

This would reduce a lot of code and reduce the need for many of the functions below too